### ISSUE TITLE
Origin filtering

### ISSUE THREAD

**Issue by Radiergummi** [Radiergummi/phoenix#1]
We need an easy way of assigning origins to parsers. I thought of a callback provided by all parsers that is run before passing it an origin. To give a simple example: Parsing files based on extension. The `JSDocParser` should only receive JS files, the `ReactTemplateParser` only JSX files. To make this work, all Readers should provide some meta information about the read file, such as name and path for file system based readers. This gets more complicated with non-file based Readers however this is a thing I definitely want to see being possible with Phoenix.  
Maybe it would be enough to have the reader create a `type` attribute on the files, retrieved from whatever information is available.

TL;DR:
  - Readers provide an `accepts(filetype: string): boolean` method
 - `filetype` is a common format, ideally a file extension or type name that Parser implementations need to deal with
- Module specifications are a top priority requirement to enable developers to write their own

**Comment by Radiergummi** [comment]
Fixed in 2f87adf4064c101207f3748f05899201a765a7de
